airflow/providers/docker/operators/docker.py

```diff
@@ -343,13 +345,20 @@ def hook(self) -> DockerHook:
             assert_hostname=self.tls_hostname,
             ssl_version=self.tls_ssl_version,
         )
-        return DockerHook(
-            docker_conn_id=self.docker_conn_id,
-            base_url=self.docker_url,
-            version=self.api_version,
-            tls=tls_config,
-            timeout=self.timeout,
-        )
+        for url in self.docker_url:
+            hook = DockerHook(
+                docker_conn_id=self.docker_conn_id,
+                base_url=url,
+                version=self.api_version,
+                tls=tls_config,
+                timeout=self.timeout,
+            )
+            try:
+                hook.get_conn()
+                return hook
+            except Exception as e:
+                self.log.error("Failed to establish connection to Docker host %s: %s", url, e)
+        raise Exception("Failed to establish connection to Docker host.")
```

Comment:
Do not raise Exception, it might be a problem in the future because for some test it would be required to catch `Exception` which is too broad, better use here AirflowException, which also is broad, but at least it Aitflow specific error

Reply:
How about `DockerException` ? I used this to fix it.

Reply:
Maybe AirflowException or python's builtin, if we could better avoid third party exceptions